#!/bin/bash

#this script (map10x2.sh) is used for the final step of short read polishing. This step
#requires the output of the script map10x1.sh to be trimmed at the overlapping ends. This
#can be achieved using mummer or BLAST to determine the coordinates for trimming. Ends
#should be trimmed leaving about 100 bp of overlapping ends, in order to achieve a good
#alignment, and those overlapping ends should be removed from the final assembly.

#it requires the following software (and their dependencies) installed:
#bowtie2/2.1.0, samtools/1.7

#reads are aligned to the reference, Similarly to script map10x1.sh, a final round of
#freebayes and bcftools consensus is required to obtain the polished contig using the 
#aligned outcome of the script (this step is currently not included in the script).

#required positional arguments are:
#the species name (e.g. Calypte_anna)
#the VGP species ID (e.g. bCalAnn1)
#the filename of the mitocontig generated by the script map10x1.sh after trimming.

set -e

SPECIES=$1
ABBR=$2
CONTIG=$3

cd ${SPECIES}/${ABBR}_10x2

#align
bowtie2-build ${CONTIG} ${ABBR}
bowtie2 -x ${ABBR} -1 ../${ABBR}_10x1/fq/aligned_${ABBR}_all_1.fq -2 ../${ABBR}_10x1/fq/aligned_${ABBR}_all_2.fq -p 16 --no-mixed | samtools view -bSF4 - > "aligned_${ABBR}_all_trimmed.bam"

#sort and index the alignment
samtools sort aligned_${ABBR}_all_trimmed.bam -o aligned_${ABBR}_all_trimmed_sorted.bam -@ 16
samtools index aligned_${ABBR}_all_trimmed_sorted.bam